epistemix_platform/use_cases/write_to_local.py

```
@@ -0,0 +1,154 @@
+    # Log if overwriting
+    if file_path.exists():
+        logger.info(f"Overwriting existing file: {file_path}")
+
+    # Determine write mode based on content type
+    if content.content_type == ContentType.ZIP_ARCHIVE:
+        # For ZIP archives, the raw_content should be base64 encoded binary data
+        if not _validate_base64(content.raw_content):
+            raise ValueError(
+                f"ZIP archive content is not valid base64: content starts with "
+                f"'{content.raw_content[:50]}...'"
+            )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)